src/recipe/test/unit/recipe.service.spec.ts

```diff
@@ -17,37 +21,46 @@ const newRecipe = {
 
 describe('Recipe Service', () => {
   let recipeService: RecipeService;
+  let commandBus: CommandBus;
+  let queryBus: QueryBus;
+  let userService: UserService;
+  let testUser: User;
   let prismaService: PrismaService;
 
-  beforeEach(async () => {
+  beforeAll(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [RecipeService, PrismaService],
+      imports: [AppModule, CqrsModule],
     }).compile();
 
+    await module.createNestApplication().init();
     recipeService = module.get<RecipeService>(RecipeService);
+    commandBus = module.get<CommandBus>(CommandBus);
+    queryBus = module.get<QueryBus>(QueryBus);
+    userService = module.get<UserService>(UserService);
     prismaService = module.get<PrismaService>(PrismaService);
+
+    testUser = await userService.generateAccount(
+      process.env.TEST_NAME,
+      process.env.TEST_PASSWORD,
+      Role.USER,
+    );
   });
 
-  it('should create a recipe', async () => {
-    const mockResult = {
-      id: 1,
-      authorId: 1,
-      name: 'Dumplings',
-      description: 'Easy dumplings recipe',
-      imageURL: 'imageURL',
-      preparing: [{ id: 1, step: 'add flour', order: 1 }],
-      ingredients: [{ id: 1, name: 'flour', amount: 'spoon' }],
-    } as any;
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
 
-    const prismaCreate = jest
-      .spyOn(prismaService.recipe, 'create')
-      .mockResolvedValue(mockResult);
+  afterAll(async () => {
+    await prismaService.recipe.deleteMany({});
+    await prismaService.user.deleteMany();
```

Comment:
I think you can do it in parallel using Promise.all:

```suggestion
    await Promise.all([prismaService.recipe.deleteMany({}),  prismaService.user.deleteMany()];
```

also is there a reason for `{}` in `recipe.deleteMany` or the other way - lack of `{}` in `user.deleteMany()`?

Reply:
also I think you should clear database after each test, so in `afterEach`, not after all tests in `afterAll`